api/v1alpha1/virtualmachineimage_conversion.go

```diff
@@ -455,9 +458,19 @@ func (dst *ClusterVirtualMachineImage) ConvertFrom(srcRaw conversion.Hub) error
 		return err
 	}
 
+	dst.Status.ContentLibraryRef = readContentLibRefConversionAnnotation(src)
+
 	return nil
 }
 
+func readContentLibRefConversionAnnotation(from ctrl.Object) (objRef *corev1.TypedLocalObjectReference) {
+	if data, ok := from.GetAnnotations()[v1alpha2.VMIContentLibRefAnnotation]; ok {
+		objRef = &corev1.TypedLocalObjectReference{}
+		_ = json.Unmarshal([]byte(data), objRef)
+	}
+	return
+}
+
```

Comment:
Instead of throwing an error and failing the conversion, we should not set the content library reference in the status field.